### PR TITLE
fix: [CDS-39348]: Adding logs to debug instance sync

### PR DESCRIPTION
**Pull request by vikyathharekal**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34837)

### CODE REVIEW

**Comment by vikyathharekal** [comment]
trigger messagecheck

**Comment by mbatchelor** [comment]
trigger messagecheck

**Comment by mbatchelor** [comment]
trigger messagecheck

File: 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java [hunk 1]
```diff
@@ -106,9 +109,12 @@ public void processInstanceSyncForNewDeployment(DeploymentEvent deploymentEvent)
             instanceSyncPerpetualTaskInfoDTO = instanceSyncPerpetualTaskInfoService.save(
                 prepareInstanceSyncPerpetualTaskInfoDTO(deploymentSummaryDTO, perpetualTaskId));
```
Comment (shubham-in10se) [inline 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java:110]: log the newly saved PerpetualTask DTO

File: 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java [hunk 2]
```diff
@@ -167,6 +173,7 @@ public void processInstanceSyncByPerpetualTask(String accountIdentifier, String
 
       InstanceSyncPerpetualTaskInfoDTO instanceSyncPerpetualTaskInfoDTO =
           instanceSyncPerpetualTaskInfoDTOOptional.get();
+      log.info("instanceSyncPerpetualTaskInfoDTO: {}", instanceSyncPerpetualTaskInfoDTO);
```
Comment (shubham-in10se) [inline 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java:176]: log all the inputs to `public void processInstanceSyncByPerpetualTask(String accountIdentifier, String perpetualTaskId,
      InstanceSyncPerpetualTaskResponse instanceSyncPerpetualTaskResponse)` as well

File: 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java [hunk 3]
```diff
@@ -79,6 +79,8 @@ public void processInstanceSyncForNewDeployment(DeploymentEvent deploymentEvent)
     int retryCount = 0;
```
Comment (shubham-in10se) [inline 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java:79]: log the whole of `DeploymentEvent` coming in as input

File: 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java [hunk 4]
```diff
@@ -229,12 +236,23 @@ Map<OperationsOnInstances, List<InstanceDTO>> handleInstanceSync(
...
+    serverInstanceInfoList.forEach(serverInstanceInfo -> log.info("serverInstanceInfo: {}", serverInstanceInfo));
+    log.info("InstanceSyncPerpetualTaskInfoDTO: {}", instanceSyncPerpetualTaskInfoDTO);
+    log.info("InfrastructureMappingDTO: {}", infrastructureMappingDTO);
+    log.info("isNewDeploymentSync: {}", isNewDeploymentSync);
+
     List<InstanceDTO> instancesInDB = instanceService.getActiveInstancesByInfrastructureMappingId(
         infrastructureMappingDTO.getAccountIdentifier(), infrastructureMappingDTO.getOrgIdentifier(),
         infrastructureMappingDTO.getProjectIdentifier(), infrastructureMappingDTO.getId());
+
+    instancesInDB.forEach(instance -> log.info("Instance in DB: {}", instance.getInstanceInfoDTO()));
```
Comment (shubham-in10se) [inline 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java:249]:
```suggestion
    instancesInDB.forEach(instance -> log.info("Instance in DB: {}", instance));
```

File: 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java [hunk 5]
```diff
@@ -306,6 +329,8 @@ private void prepareInstancesToBeDeleted(Map<OperationsOnInstances, List<Instanc
     Sets.SetView<String> instancesToBeDeleted =
         Sets.difference(instancesInDBMap.keySet(), instanceInfosFromServerMap.keySet());
```
Comment (shubham-in10se) [inline 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java:330]: log `instancesInDBMap.keySet(), instanceInfosFromServerMap.keySet()`

File: 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java [hunk 6]
```diff
@@ -332,6 +359,8 @@ private void prepareInstancesToBeUpdated(AbstractInstanceSyncHandler instanceSyn
     Sets.SetView<String> instancesToBeUpdated =
         Sets.intersection(instanceInfosFromServerMap.keySet(), instancesInDBMap.keySet());
```
Comment (shubham-in10se) [inline 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java:360]: log `instanceInfosFromServerMap.keySet(), instancesInDBMap.keySet()`

**Review by shubham-in10se — COMMENTED** [review]
General Comment:
the logs that are finally printed in stackdriver do not contain the line number of the code. It might be difficult to recognise where the log is coming from in the file, with so many similar logs coming in 😓 
Let's also try to add line number in the log statement itself, or any other identifier.

**Review by shubham-in10se — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34837)* status: :shipit: complete! ✓

**Comment by vikyathharekal** [comment]
trigger messagecheck

**Review by shubham-in10se — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34837)* status: :shipit: complete! ✓

**Pull request merged**